design/design.md

```diff
@@ -0,0 +1,175 @@
+
+![createShip diagram](img/createShip.png)
+
+
+### Move a Ship:
+Updates the `pos_x`, `pos_y` and `fuel` datum fields of the `ShipState` UTxO by adding the `delta_x` and `delta_y` values specified in the redeemer, and subtracting the fuel amount needed for the displacement.
+
+![moveShip diagram](img/moveShip.png)
+
+
+### Gather Fuel:
+Updates the `fuel` datum field of both the `ShipState` and `PelletState` UTxOs, adding the `amount` (specified in the redeemer) from the first and subtracting it from the latter.
+
+![gatherFuel diagram](img/gatherFuel.png)
+
+
+### Collect Rewards:
+Subtracts from the `Pot` UTxO 50% of the ada value, and pays that amount to the owner of the ship that reached the pot, together with the min ada locked in the `ShipState` UTxO. The ship token is burnt.
+
+![collect diagram](img/collect.png)
+
+
+### Quit Game:
+Pays the min ada locked in the `ShipState` UTxO back to the ship owner and burns the ship token.
+
+![quit diagram](img/quit.png)
+
+
+## Validators & Minting Policies
+
+### Pot validator:
+* Params: admin token.
+
+#### *AddNewPlayer Redeemer*
+* `Pot` output value equals input value plus the inscription fee.
+* adminToken is in the input.
+* datum doesn't change.
+
+#### *Collect Redeemer (includes address of the player that reached the pot)*
+* ship token is present in some input.
+* `Pot` output value has at most 50% adas less than input value.
+
+### PelletState validator:
+* Params: admin token.
+
+#### *Gather Redeemer (includes gathering amount)*
+* ship token is present in some input.
+* there is a `ShipState` input with the same x and y datum coordinates as the `PelletState` UTxO.
+* the amount specified is not greater than the fuel available in the pellet.
+* the amount specified is subtracted from the output `PelletState` fuel datum field, and the other fields remain unchanged.
+
+### ShipState validator:
+* Params: admin token, `Pot` validator address and `PelletState` validator address.
+
+#### *MoveShip Redeemer (includes delta_x and delta_y displacements)*
+* there is a single `ShipState` input.
+* there is a single `ShipState` output.
+* the `PilotToken` is present in an input.
+* the `ShipState` input has enough fuel to move the desired delta.
+* the pilot_token datum field is not changed.
+* the x and y output datum values are updated as the previous ones (input values) plus the corresponding deltas.
+* the output fuel datum field equals the input fuel minus the fuel required for the displacement.
+* the distance advanced doesn't exceed the `MAX_SHIP_MOVEMENT_PER_TX`.
+* the tx is signed by the ship owner.
+
+#### *Gather Redeemer (includes gathering amount)*
+* the amount specified plus the fuel before charging does not exceed the ship's capacity.
+* the amount specified is added to the output `ShipState` fuel datum field, and the other fields remain unchanged.
+* the `ShipState` output value is the same as the input.
+
+#### *Collect Redeemer*
```

Comment:
you forgot the fundamental check: ship position is (0,0)

Reply:
Also: ship token is burnt (if not here, it should be checked in the pot validator).
Also: only ONE ship state is in the input.

Reply:
Also: pilot token is present.